File: library/java/c3/stack.java
public class Stack <T> {
	private class StackNode <T> {
		private T data;
		private StackNode<T> next;

		public StackNode(T element) {
			this.data = element;
			this.next = null;			
		}
	}

	private StackNode<T> top;

	public Stack() {
		this.top = null;
	}	

	public void push(T data){
		StackNode<T> myNode = new StackNode<T>(data);
		myNode.next = this.top;
		this.top = myNode;
	}

	public T peek(){
		return this.top.data;
	}

	public T pop(){		
		T data = null;
		if (this.top != null){
			data = this.top.data;
			this.top = this.top.next;
		}				
		return data;
	}

	public static void main(String [] args){
		Stack<Integer> myStack = new Stack<Integer>();
		myStack.push(7);
		myStack.push(4);
		myStack.push(2);
		myStack.push(1);
		System.out.println("P");
		System.out.println(myStack.peek());
	}	
}

File: python/data_structures/stack.py
class Node:
	def __init__(self, data, next):
		self.data = data
		self.next = next

class Stack:
	def __init__(self):
		self.head = None
	
	def push(self, data):
		self.head = Node(data, self.head)

	def pop(self):
		data = self.head.data
		self.head = self.head.next
		return data 


class StackArray:
	
	def __init__(self):
		self.data = []

	def push(self, item):
		self.data.append(item)

	def pop(self):
		return self.data.pop()

	def peek(self):
		return self.data[-1]        